src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs

```diff
@@ -46,6 +47,10 @@ protected ModelMetadata(ModelMetadataIdentity identity)
         Identity = identity;
 
         InitializeTypeInformation();
+        if (RuntimeFeature.IsDynamicCodeSupported)
+        {
+            InitializeDynamicTypeInformation();
```

Comment:
This will cause a trimming warning becuase:

```
    [RequiresUnreferencedCode("Using ModelMetadata with dynamic dependencies enabled is not trim compatible.")]
    [RequiresDynamicCode("Using ModelMetadata with dynamic dependencies enabled is not native AOT compatible.")]
    private void InitializeDynamicTypeInformation()
```

InitializeDynamicTypeInformation is marked as both "unsafe for trimming" and "unsafe for AOT". But here we are only protecting against the latter. When just trimming an app (which is an important scenario) will cause warnings here because `IsDynamicCodeSupported == true`, but we are calling into a function that is marked as unsafe for trimming.

Reply:
If you aren't getting a warning now, it would be good to understand why this isn't getting caught by our checks.